contracts/CrossStake.sol

```diff
@@ -0,0 +1,99 @@
+pragma solidity 0.6.4;
+
+import "./System.sol";
+import "./interface/ICrossChain.sol";
+import "./interface/IParamSubscriber.sol";
+import "./interface/IApplication.sol";
+import "./lib/SafeMath.sol";
+import "./lib/RLPDecode.sol";
+
+contract CrossStake is System, IParamSubscriber, IApplication {
+  using SafeMath for uint256;
+  using RLPDecode for *;
+
+  uint8 constant public CROSS_STAKE_CHANNELID = 0x10;
+
+  // Cross-Chain Stake Event type
+  uint8 public constant EVENT_STAKE = 0x00;
+  uint8 public constant EVENT_UNSTAKE = 0x01;
+  uint8 public constant EVENT_CLAIM_REWARD = 0x02;
+  uint8 public constant EVENT_CLAIM_UNSTAKE = 0x03;
+  uint8 public constant EVENT_RESTAKE = 0x04;
+
+  //TODO
+  uint256 public constant INIT_ORACLE_RELAYER_FEE = 0;
+  uint256 public constant INIT_BSC_RELAYER_FEE = 0;
+
+  uint256 public oracleRelayerFee;
+  uint256 public BSCRelayerFee;
+
+  modifier initRelayerFee() {
+    if (oracleRelayerFee == 0) {
+      oracleRelayerFee = INIT_ORACLE_RELAYER_FEE;
+      BSCRelayerFee = INIT_BSC_RELAYER_FEE;
+    }
+    _;
+  }
+
+  function stake(address validator) payable initRelayerFee {
+    uint256 amount = msg.value;
+    require(amount > oracleRelayerFee, "Send value cannot cover the relayer fee");
+
+    bytes memory msgBytes;
+    msgBytes = abi.encode(EVENT_STAKE, msg.sender, amount-oracleRelayerFee, validator);
```

Comment:
Please do not use abi encode but still follow rlp encode just like other cross chain app.

Reply:
I will suggest the package as:
```
{
  EventType: uint8,
  EventInfo:    bytes, // rlp encoded.
}
```

Reply:
The encode method has been fixed. But I don't understand the meaning of 
```
{
  EventType: uint8,
  EventInfo:    bytes, // rlp encoded.
}
```